tests/unit/ui/components/TestOverlay.js

```diff
@@ -1,73 +1,107 @@
+    const html = <div>a child div</div>;
+    const props = getProps(otherProps);
+    const root = shallowUntilTarget(
+      <Overlay {...props}>{children || html}</Overlay>,
+      OverlayBase,
+    );
+
+    // Apply initial UI state.
+    applyUIStateChanges({ root, store: props.store });
+    return root;
+  };
+
+  it('is hidden by default', () => {
     const root = render();
-    expect(root.overlayContainer).toBeTruthy();
-    expect(root.overlayBackground).toBeTruthy();
-    expect(root.overlayContainer.tagName).toEqual('DIV');
-    expect(root.overlayContainer.className).toContain('Overlay');
+    expect(root).not.toHaveClassName('Overlay--visible');
   });
 
-  it('renders extra className if provided', () => {
-    const root = render({ className: 'I-am-so-over-it' });
-    expect(root.overlayContainer.className).toContain('Overlay');
-    expect(root.overlayContainer.className).toContain('I-am-so-over-it');
-  });
+  it('becomes visible when its mounted with visibleOnLoad prop as true', () => {
+    const root = render({ visibleOnLoad: true });
 
-  it('renders children', () => {
-    const root = render({ children: 'hello' });
-    expect(root.overlayContents).toBeTruthy();
-    expect(root.overlayContents.textContent).toContain('hello');
+    expect(root).toHaveClassName('Overlay');
+    expect(root).toHaveClassName('Overlay--visible');
+    expect(root.find('.Overlay-background')).toHaveLength(1);
+    expect(root.find('.Overlay-contents')).toHaveLength(1);
   });
 
-  it('is hidden by default', () => {
-    const root = render();
-    expect(root.overlayContainer.className).not.toContain('Overlay--visible');
+  it('becomes visible once the visibleOnLoad prop of true is passed', () => {
+    const { store } = dispatchClientMetadata();
+
+    const root = render({ store });
+
+    expect(root).not.toHaveClassName('Overlay--visible');
+
+    root.setProps({ visibleOnLoad: true });
+
+    applyUIStateChanges({ root, store });
+
+    expect(root).toHaveClassName('Overlay--visible');
   });
 
-  it('is visible when the `visibleOnLoad` prop is passed', () => {
-    const root = render({ visibleOnLoad: true });
-    expect(root.overlayContainer.className).toContain('Overlay--visible');
+  it('renders extra className if provided', () => {
+    const className = 'I-am-so-over-it';
+    const root = render({ visibleOnLoad: true, className });
+    expect(root).toHaveClassName(className);
   });
 
-  it('hides when you click the background', () => {
-    const root = render({ visibleOnLoad: true });
-    Simulate.click(root.overlayBackground);
-    expect(root.overlayContainer.className).not.toContain('Overlay--visible');
+  it('renders children', () => {
+    const text = 'overriding div text..';
+    const html = <div>{text}</div>;
+    const root = render({ visibleOnLoad: true, children: html });
+
+    expect(
+      root
+        .find('.Overlay-contents')
+        .children()
+        .html(),
+    ).toContain(text);
```

Comment:
I am still not sure about this. I believe I tried a few a things here. let me know if this works...